rollouts/controllers/rollout_controller.go

```diff
@@ -154,10 +154,29 @@ func (r *RolloutReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	packageDiscoveryClient := r.getPackageDiscoveryClient(req.NamespacedName)
 
-	err = r.reconcileRollout(ctx, &rollout, strategy, packageDiscoveryClient)
+	targetClusters, err := r.store.ListClusters(ctx, &rollout.Spec.Clusters, rollout.Spec.Targets.Selector)
+	if err != nil {
+		logger.Error(err, "Failed to list clusters")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	discoveredPackages, err := packageDiscoveryClient.GetPackages(ctx, rollout.Spec.Packages)
+	if err != nil {
+		logger.Error(err, "Failed to discover packages")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	logger.Info("Discovered packages", "packagesCount", len(discoveredPackages), "packages", packagediscovery.ToStr(discoveredPackages))
+
+	allClusterStatuses, waveStatuses, err := r.reconcileRollout(ctx, &rollout, strategy, targetClusters, discoveredPackages)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
+
+	sortClusterStatuses(allClusterStatuses)
```

Comment:
Should we keep `sortClusterStatuses` inside the `reconcileRollout` that will ensure the function being tested return a stable ordered clusterstatuses ?

Reply:
sgtm! Moved back